### PR TITLE
Add Docker build job in pull requests

### PR DESCRIPTION
**Pull request by ghost**
Adds the Docker build job to all pull requests, so we don't run into a postsubmit problem with Docker like we have ATM.

Signed-off-by: Matt Wilkinson <matt@mattwilkinson.dev>

### CODE REVIEW

File: .github/workflows/docker_check.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          tags: |
+            wasmcloud/wash:alpine
+          labels: ${{ steps.meta.outputs.labels }}
+      - name: Build and push - scratch
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: "Dockerfile"
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```
Comment (brooksmtownsend) [inline .github/workflows/docker_check.yml]: This looks like an extra step

File: .github/workflows/docker_check.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            org.opencontainers.image.title=wash
+            org.opencontainers.image.description=WAsmcloud SHell
+            org.opencontainers.image.vendor=wasmCloud
+            org.opencontainers.image.source=https://github.com/${{ github.repository }}
+      - name: Build
+        id: docker_build_builder
+        uses: docker/build-push-action@v2
+        with:
+          file: "Dockerfile"
+          target: release-alpine
```
Comment (brooksmtownsend) [inline .github/workflows/docker_check.yml:42]: We should probably add an explicit `push: false` to this action just to make sure we aren't pushing on PR

I know it's the default, but I don't want to get burned by a GH action accidentally updating under us 😄

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Small request

File: .github/workflows/docker_check.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/docker_check.yml]: Removed.

File: .github/workflows/docker_check.yml (hunk 2 above)
Comment (ghost) [reply .github/workflows/docker_check.yml:42]: Done.

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**